blackbox_repository/conversion_scripts/scripts/nasbench201_import.py

```diff
@@ -0,0 +1,212 @@
+import bz2
+import pickle
+
+import matplotlib.pyplot as plt
+
+import pandas as pd
+import numpy as np
+from blackbox_repository.blackbox_tabular import serialize, BlackboxTabular
+import sagemaker_tune.search_space as sp
+from blackbox_repository.conversion_scripts.utils import repository_path
+from syne_tune.util import catchtime
+
+
+def str_to_list(arch_str):
+
+    node_strs = arch_str.split('+')
+    config = []
+    for i, node_str in enumerate(node_strs):
+        inputs = list(filter(lambda x: x != '', node_str.split('|')))
+        for xinput in inputs: assert len(xinput.split('~')) == 2, 'invalid input length : {:}'.format(xinput)
+        inputs = (xi.split('~') for xi in inputs )
+
+        config.extend(op for (op, IDX) in inputs)
+
+    return config
+
+
+def convert_dataset(data, dataset):
+
+    hp_cols = [f"hp_x{i}" for i in range(6)]
+
+    hps = dict()
+
+    for h in hp_cols:
+        hps[h] = []
+
+    n_hps = data['total_archs']
+
+    for i in range(n_hps):
+        config = str_to_list(data['arch2infos'][i]['200']['arch_str'])
+
+        for j, hp in enumerate(config):
+            hps[f"hp_x{j}"].append(hp)
+
+    hyperparameters = pd.DataFrame(
+        data=hps,
+        columns=hp_cols
+    )
+
+    objective_names = [
+        'valid_error',
+        'train_error',
+        'runtime',
+        'latency',
+        'flops',
+        'params',
+
+    ]
+
+    fidelity_values = np.arange(1, 201)
+    n_fidelities = len(fidelity_values)
+    n_objectives = len(objective_names)
+    n_seeds = 3
+
+    objective_evaluations = np.empty((n_hps, n_seeds, n_fidelities, n_objectives)).astype('float32')
+
+    def save_objective_values_helper(name, values):
+        assert values.shape == (n_hps, n_seeds, n_fidelities)
+
+        name_index = dict(zip(
```

Comment:
Nit, but the dict can be taken out or passed as argument: it is constructed every time but does not change (wont change performance but might make things clearer).

Reply:
I also think this would be easier to read:
```suggestion
        name_index = {name: i for i, name in enumerate(objective_names)}
```